src/main/java/uk/gov/hmcts/darts/authentication/component/SoapRequestInterceptor.java

```diff
@@ -202,4 +218,33 @@ private void logPayloadMessage(String messagePrefix, WebServiceMessage message)
             log.error("Failed to write SOAP message", ex);
         }
     }
+
+    private void verifyBasicAuthorisationRequestIsAllowed(String userName, SaajSoapMessage message) {
+        if (isContextRegistryRequest(message)) {
+            return;
+        }
+
+        if (!securityProperties.getExternalServiceBasicAuthorisationWhitelist().contains(userName)) {
+            throw new BasicAuthorisationFailedException(userName);
+        }
+    }
+
+    private boolean isContextRegistryRequest(SaajSoapMessage message) {
```

Comment:
we need to performance test this. Is quite heavy. Could be quicker to look for the context registry namespace in the soap request body

Reply:
ok, I wil ask the perf team to add this to their backlog of things to performance test